addons/web/static/src/views/form/form_controller.js

```diff
@@ -159,7 +159,15 @@ export class FormController extends Component {
 
         const beforeExecuteAction = (clickParams) => {
             if (clickParams.special !== "cancel") {
-                return this.model.root.save({ stayInEdition: true });
+                const saveProm = this.model.root.save({ stayInEdition: true });
+                if (this.props.saveRecord) {
+                    saveProm.then(() => {
+                        this.props.saveRecord(this.model.root);
+                    });
+                }
+                return saveProm;
```

Comment:
if there is a saveRecord prop it is supposed to do the save (and so we shouldn't call this.model.root.save in that case, as seen in the other place we call props.saveRecord, we pass the model root to the saveRecord prop and it can do that itself, optionally doing things before and after), you should also forward the params (in this case the `{ stayInEdition: true }`